crates/db_schema/src/source/activity.rs

```diff
@@ -10,13 +76,60 @@ pub struct SentActivity {
   pub data: Value,
   pub sensitive: bool,
   pub published: chrono::NaiveDateTime,
+  #[diesel(deserialize_as = ArrayToHashSet<DbUrl>)]
+  pub send_inboxes: HashSet<DbUrl>,
+  #[diesel(deserialize_as = ArrayToHashSet<CommunityId>)]
+  pub send_community_followers_of: HashSet<CommunityId>,
+  pub send_all_instances: bool,
+  pub actor_type: ActorType,
+  pub actor_apub_id: Option<DbUrl>,
+}
+
+// wrapper to remove optional from array values and convert to hashset
+pub struct ArrayToHashSet<T>(HashSet<T>);
+
+impl<DB, T1, T2> Queryable<Array<Nullable<T2>>, DB> for ArrayToHashSet<T1>
+where
+  DB: Backend,
+  T1: FromSql<T2, DB> + Hash + Eq,
+  Vec<std::option::Option<T1>>: FromSql<Array<Nullable<T2>>, DB>,
+{
```

Comment:
Scares me a bit, I really hope you're not doing filtering of this array column in the code. Postgres should be handling that with a proper bridge table.